Source/WebCore/page/Location.cpp

```diff
@@ -64,11 +65,10 @@ const Frame* Location::frame() const
 
 const URL& Location::url() const
 {
-    RefPtr localFrame = dynamicDowncast<LocalFrame>(frame());
-    if (!localFrame)
-        return aboutBlankURL();
+    ASSERT(m_window);
+    ASSERT(downcast<LocalDOMWindow>(*m_window).document());
```

Comment:
This downcast isn't safe, as far as I know, this can be a RemoteDOMWindow as of recently. It should be a dynamicDowncast as suggested earlier.

Reply:
OK, now a dynamicDowncast